File: src/main/java/nz/co/searchwellington/controllers/models/GeotaggedNewsitemExtractor.java
package nz.co.searchwellington.controllers.models;

import java.util.List;

import nz.co.searchwellington.model.frontend.FrontendFeedNewsitem;
import nz.co.searchwellington.model.frontend.FrontendNewsitem;

import org.springframework.stereotype.Component;

import com.google.common.collect.Lists;

@Component
public class GeotaggedNewsitemExtractor {
	
	public List<FrontendNewsitem> extractGeotaggedItems(List<FrontendNewsitem> feedNewsitems) {
		List<FrontendNewsitem> geotaggedFeedNewsitems = Lists.newArrayList();
		for (FrontendNewsitem feedNewsitem : feedNewsitems) {
			if (feedNewsitem.getPlace() != null) {
				geotaggedFeedNewsitems.add(feedNewsitem);
			}
		}
		return geotaggedFeedNewsitems;
	}

	public List<FrontendNewsitem> extractGeotaggedItemsFromFeedNewsitems(List<FrontendFeedNewsitem> feedNewsitems) {	// TODO really Java? why are we not allowed to just call the above?
		List<FrontendNewsitem> geotaggedFeedNewsitems = Lists.newArrayList();
		for (FrontendNewsitem feedNewsitem : feedNewsitems) {
			if (feedNewsitem.getPlace() != null) {
				geotaggedFeedNewsitems.add(feedNewsitem);
			}
		}
		return geotaggedFeedNewsitems;
	}
	
}

File: src/main/scala/nz/co/searchwellington/controllers/models/GeotaggedNewsitemExtractor.scala
package nz.co.searchwellington.controllers.models

import nz.co.searchwellington.model.frontend.FrontendResource
import org.springframework.stereotype.Component

@Component class GeotaggedNewsitemExtractor {

  def extractGeotaggedItems(feedNewsitems: Seq[FrontendResource]): Seq[FrontendResource] = {
    feedNewsitems.filter(_.geocode.nonEmpty).reverse  // TODO pushes the most recent to last so it's rendered top of stack for overlapping points; view concern in the wrong place
  }

}